src/global-messages/GlobalMessage.tsx

```diff
@@ -9,13 +9,19 @@ import {
 } from '@atb/global-messages/types';
 import {getTextForLanguage} from '@atb/translations';
 import {useNow} from '@atb/utils/use-now';
+import {isWithinTimeRange} from '@atb/global-messages/is-within-time-range';
 
 type Props = {
   globalMessageContext?: GlobalMessageContextType;
   style?: StyleProp<ViewStyle>;
+  showAllMessages?: boolean;
```

Comment:
I think this can be improved. Generally we do not want two different parameters affecting the same logic and overriding each other. As of now setting `showAllMessages` makes the value in `globalMessageContext` redundant.

A more precise typing could be something like this:
```ts
type Props = {
  globalMessageContext: GlobalMessageContextType;
  style?: StyleProp<ViewStyle>;
  showAllMessages: undefined | false;
} |  {
  style?: StyleProp<ViewStyle>;
  showAllMessages: true;
}
```

With this typing you say that if `showAllMessages` is `true` then you don't need to (and can't) send in a `globalMessageContext`. Also if `showAllMessages` is not `true`, then you need to send in a `globalMessageContext`.

Another option is to differ between logic regarding context and logic regarding whether dismissed messages should be shown. That can be done by having a typing that is something like this:
```ts
type Props = {
  globalMessageContext: GlobalMessageContextType | 'all'; // Note 'all' added here
  style?: StyleProp<ViewStyle>;
  includeDismissed?: boolean;
}
```

Personally I think the latter suggestion is a little more precise, since it is clearer distinction between what logic the two parameters `globalMessageContext` and `includeDismissed` affects.